### ISSUE TITLE
multiline-ternary doesn't check parentheses

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13195]
**Tell us about your environment**

* **ESLint Version:** v7.0.0-alpha.3
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhXG4pID8gYlxuICA6IGM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

```js
/*eslint multiline-ternary: ["error", "always"] */

(a
) ? b
  : c;
```

```bash
eslint index.js
```

**What did you expect to happen?**

It might be expected that parentheses should be treated as a part of the node, so in this case the rule should report that there's a missing newline between test and consequent.

Some rules work that way (e.g., `array-element-newline`), some don't (`function-call-argument-newline`), so I'm not sure if this is a bug.

**What actually happened? Please include the actual, raw output from ESLint.**

no errors

**Are you willing to submit a pull request to fix this bug?**

yes

The example was with `"always"`, but the same issue exists with `"always-multiline"` and `"never"`.

**Comment by anikethsaha** [comment]
I think this should be an error as well, right !

```js
(a) 
  ? (b
    ): c; // <== error
```

**Comment by mdjermanovic** [comment]
Yes, if this is a bug the same logic would apply to the check between consequent and alternate..

**Comment by kaicataldo** [comment]
I personally think this should be classified as a bug, but agree that we should be consistent between rules.

**Comment by mdjermanovic** [comment]
I'm working on this.

**Comment by jsphstls** [comment]
This change is causing some problems.

[Here's the first problem from above](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhXG4pID8gYlxuICA6IGM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19):
```
(a
) ? b
  : c;
```

[Here's what this rule allows after the MR](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhXG4pXG4gID8gYlxuICA6IGM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). The ternary has not really become more readable.:
```
(a
)
  ? b
  : c;
```

[This should be the only solution](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKVxuICA/IGJcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
(a)
  ? b
  : c;
```

[Another mentioned example](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYlxuICAgICk6IGM7IC8vIDw9PSBlcnJvciIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
(a) 
  ? (b
    ): c; // <== error
```

[Fix allowed by the rule change](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYlxuICAgIClcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
(a) 
  ? (b
    )
  : c;
```

[A better fix](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYilcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
(a) 
  ? (b)
  : c;
```

This is affecting me because I cannot upgrade my core Eslint version without adding a line to every otherwise readable ternary.

[This is now invalid](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbmFcbiAgPyAoXG4gICAgYlxuICApIDogKFxuICAgIGNcbiAgKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
a
  ? (
    b
  ) : (
    c
  )
```

[Here's how I must now fix each of those cases](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbmFcbiAgPyAoXG4gICAgYlxuICApXG4gIDogKFxuICAgIGNcbiAgKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
```
a
  ? (
    b
  )
  : ( // extra line here
    c
  )
```

So why have the multiline parens? Because it handles multiline consequents while keeping alignment. This used to be allowed:
```
isSomething
  ? (
    <SomeJsx
      someAttr
      someOtherAttr
    />
  ) : (
    <SomeOtherJsx
      someAttr
      someOtherAttr
    />
  )
```

Now I am required to do this:
```
isSomething
  ? (
    <SomeJsx
      someAttr
      someOtherAttr
    />
  )
  : ( // extra line added here
    <SomeOtherJsx
      someAttr
      someOtherAttr
    />
  )
```

**Comment by mdjermanovic** [comment]
@jsphstls thanks for the suggestion and the detailed examples!

> 
> [Here's the first problem from above](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhXG4pID8gYlxuICA6IGM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19):
> 
> ```
> (a
> ) ? b
>   : c;
> ```
> 
> [Here's what this rule allows after the MR](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhXG4pXG4gID8gYlxuICA6IGM7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). The ternary has not really become more readable.:
> 
> ```
> (a
> )
>   ? b
>   : c;
> ```
> 
> [This should be the only solution](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKVxuICA/IGJcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> (a)
>   ? b
>   : c;
> ```

I think this is correct behavior as far as this rule is concerned.

This rule doesn't control newlines in parentheses and doesn't control newlines inside the consequent/alternate expressions, so both the second and the third example are valid code for this rule, because `)` (as the end of the consequent)  and `b` are on separate lines.

We didn't change this behavior, both examples were already valid in the previous version (ESLint v7.1.0).

> [Another mentioned example](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYlxuICAgICk6IGM7IC8vIDw9PSBlcnJvciIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> (a) 
>   ? (b
>     ): c; // <== error
> ```
> 
> [Fix allowed by the rule change](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYlxuICAgIClcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> (a) 
>   ? (b
>     )
>   : c;
> ```
> 
> [A better fix](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbihhKSBcbiAgPyAoYilcbiAgOiBjOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> (a) 
>   ? (b)
>   : c;
> ```

Same as above. Both fixes should be valid for this particular rule, and they used to be valid before this change.

> This is affecting me because I cannot upgrade my core Eslint version without adding a line to every otherwise readable ternary.

I hope we'll be able to add autofixing to this rule soon, there's ongoing PR #12982.

> [This is now invalid](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbmFcbiAgPyAoXG4gICAgYlxuICApIDogKFxuICAgIGNcbiAgKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> a
>   ? (
>     b
>   ) : (
>     c
>   )
> ```
> 
> [Here's how I must now fix each of those cases](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLXRlcm5hcnk6IFtcImVycm9yXCIsIFwiYWx3YXlzXCJdICovXG5cbmFcbiAgPyAoXG4gICAgYlxuICApXG4gIDogKFxuICAgIGNcbiAgKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==):
> 
> ```
> a
>   ? (
>     b
>   )
>   : ( // extra line here
>     c
>   )
> ```
> 
> So why have the multiline parens? Because it handles multiline consequents while keeping alignment. This used to be allowed:
> 
> ```
> isSomething
>   ? (
>     <SomeJsx
>       someAttr
>       someOtherAttr
>     />
>   ) : (
>     <SomeOtherJsx
>       someAttr
>       someOtherAttr
>     />
>   )
> ```
> 
> Now I am required to do this:
> 
> ```
> isSomething
>   ? (
>     <SomeJsx
>       someAttr
>       someOtherAttr
>     />
>   )
>   : ( // extra line added here
>     <SomeOtherJsx
>       someAttr
>       someOtherAttr
>     />
>   )
> ```

This is a matter of preference, of course, but I'd argue that consequent and alternate seem to be more consistently aligned in the fixes than in the original versions: `?` is aligned with `:`, and the opening parentheses are also aligned.
